### PR TITLE
excluding classes

### PR DESCRIPTION
**Pull request by mkohram**
Allows classes to be excluded from the outside click behavior.

### CODE REVIEW

**Comment by nucleartide** [comment]
Thanks for the PR! I'll take a look at this tomorrow or Friday – it's getting kinda late atm. :D

File: Readme.md [hunk 1]
```diff
@@ -33,6 +33,11 @@ $ ember install ember-outside-click
 
 [See the demo code][3].
 
+#### Excluded classes
+
+Add the `excludedClasses` array property to either the block form or subclass to exclude classes
+from the outside click behavior.
+
```
Comment (nucleartide) [inline Readme.md:48]: Awesome, please link to the subclass example, and add an example below this line for the block form. Like:

``` hbs
{{#outside-click excludedClasses=(array 'excluded-class' 'special-toolbar')}}{{/outside-click}}
```

File: tests/dummy/app/components/drop-down/component.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ import OutsideClick from 'ember-outside-click/components/outside-click/component
 const Dropdown = OutsideClick.extend({
   open: false,
```
Comment (nucleartide) [inline tests/dummy/app/components/drop-down/component.js]: Nitpick: remove this newline please. :)

Also, I guess this was removed at some point – please add `classNames: 'Dropdown'`, so you can see the styles.

File: tests/dummy/app/styles/app.css [hunk 3]
```diff
@@ -16,3 +15,6 @@ body {
   list-style: none;
 }
 
+.an-excluded-class {
+    padding: 5px;
```
Comment (nucleartide) [inline tests/dummy/app/styles/app.css]: Nitpick: change this to two space indentation. Also, please replace the `padding` with `background-color: #eee;` – just so the click area is more visible.

File: tests/dummy/app/templates/application.hbs [hunk 4]
```diff
@@ -7,3 +7,4 @@ Click the dropdown below, then try clicking outside of the dropdown
 
 {{drop-down}}
 
+<div class="an-excluded-class">Clicking here won't close the dropdown (even though it's outside the component)</div>
```
Comment (nucleartide) [inline tests/dummy/app/templates/application.hbs]: Nitpick: format this with a 80-column limit and 2-space indentation, like:

``` hbs
<div class="an-excluded-class">
  Clicking here won't close the dropdown (even though it's outside the
  component)
</div>
```

File: addon/components/outside-click/component.js [hunk 5]
```diff
@@ -35,7 +36,14 @@ export default Component.extend({
 
   handleDown(e) {
     if (this.isDestroyed || this.isDestroying) return
-    if (!this.element.contains(e.target)) this.set('isOutside', true)
+
+    let isExcluded = this.get('excludedClasses').some((excludedClass)=>{
```
Comment (nucleartide) [inline addon/components/outside-click/component.js]: Looks good – please format this like so:

``` js
const isExcluded = this.get('excludedClasses').some(excludedClass => {
  return ` ${e.target.className} `.indexOf(` ${excludedClass} `) > -1
})
```

File: addon/components/outside-click/component.js [hunk 6]
```diff
@@ -35,7 +36,14 @@ export default Component.extend({
...
+      if ( (` ${e.target.className} `).replace(/[\n\t]/g, " ").indexOf(` ${excludedClass} `) > -1 ) return true
+
+      return false
+    });
+
+    if (!this.element.contains(e.target) && !isExcluded) this.set('isOutside', true)
```
Comment (nucleartide) [inline addon/components/outside-click/component.js]: Nitpick: format this with an 80-column limit, like:

``` js
if (!this.element.contains(e.target) && !isExcluded) {
  this.set('isOutside', true)
}
```

**Review by nucleartide — DISMISSED** [review]
Looks good! Just some minor formatting I suppose.

**Comment by mkohram** [comment]
OK. Sorry for awful formatting. Take a look.

**Comment by nucleartide** [comment]
Haha, sorry! :D I don't mean to be annoying.

**Comment by mkohram** [comment]
@nucleartide Could you look over this? I think it's close to a merge. Thanks.

**Comment by nucleartide** [comment]
Whoops, my bad! I didn't get an alert about your commits. Will look it over right away.

**Comment by nucleartide** [comment]
@mkohram released 1.1.0 and updated the demo, btw :)

**Comment by mkohram** [comment]
Thanks!

**Comment by mkohram** [comment]
@nucleartide demo doesn't seem to be updated.

**Comment by nucleartide** [comment]
@mkohram ah thanks for the catch, updated

**Pull request merged**